lib/general_widgets/expandable_textfield.dart

```diff
@@ -0,0 +1,224 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+import '../ui/shared/shared.dart';
+
+class ExpandableTextField extends StatelessWidget {
+  const ExpandableTextField({
+    Key? key,
+    required this.hintText,
+    required this.isExpanded,
+    required this.isVisible,
+    required this.focus,
+    required this.toggleExpanded,
+    required this.toggleVisibility,
+    required this.sendMessage,
+    required this.controller,
+  }) : super(key: key);
+
+  final String hintText;
+  final bool isExpanded;
+  final bool isVisible;
+  final FocusNode focus;
+  final TextEditingController controller;
+  final void Function() toggleExpanded;
+  final void Function(bool) toggleVisibility;
+  final void Function() sendMessage;
+
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: isExpanded
+          ? MediaQuery.of(context).size.height - kToolbarHeight
+          : null,
+      alignment: Alignment.bottomCenter,
+      color: Colors.white,
+      child: Column(
+        mainAxisSize: isExpanded ? MainAxisSize.max : MainAxisSize.min,
+        children: [
+          Divider(height: 0, color: Color(0xFF999999)),
+          Container(
+            alignment: Alignment.topRight,
+            child: Visibility(
+              visible: isExpanded,
+              child: GestureDetector(
+                onTap: () => toggleExpanded(),
+                child: Padding(
+                  padding: const EdgeInsets.all(8.0),
+                  child: SvgPicture.asset(
+                    'assets/icons/svg_icons/minimize.svg',
+                    color: AppColors.darkGreyColor,
+                  ),
+                ),
+              ),
+            ),
+          ),
+          Visibility(
+            visible: isExpanded,
+            child: Expanded(
+              child: _buildTextField(),
```

Comment:
Please make this a class widget and not a function widget.
Example is BuildTextField() and not _buildTextField()

Reply:
Solved